src/main/profiling_info.cpp

```diff
@@ -26,15 +26,68 @@ profiler_settings_t ProfilingInfo::DefaultOperatorSettings() {
 	};
 }
 
+profiler_settings_t ProfilingInfo::PhaseTimingsSettings() {
+	return {
+        MetricsType::ALL_OPTIMIZERS,
+        MetricsType::CUMULATIVE_OPTIMIZER_TIMING,
+	    MetricsType::PHYSICAL_PLANNER_TIMING,
+	    MetricsType::PHYSICAL_PLANNER_CREATE_PLAN_TIMING,
+	    MetricsType::PHYSICAL_PLANNER_COLUMN_BINDING_TIMING,
+	    MetricsType::PHYSICAL_PLANNER_RESOLVE_TYPES_TIMING,
+	    MetricsType::PLANNER_TIMING,
+	    MetricsType::PLANNER_BINDING_TIMING,
+	};
+}
+
+profiler_settings_t ProfilingInfo::AllSettings() {
+	auto all_settings = DefaultSettings();
+	auto optimizer_settings = GetAllOptimizerMetrics();
+	auto phase_timings = PhaseTimingsSettings();
+
+	for (auto &setting : optimizer_settings) {
+		all_settings.insert(setting);
+	}
+
+	for (auto &setting : phase_timings) {
+        all_settings.insert(setting);
+    }
+
+	return all_settings;
+}
+
+bool ProfilingInfo::IsOptimizerMetric(MetricsType metric) {
+	auto optimizers = GetAllOptimizerMetrics();
+	if (std::find(optimizers.begin(), optimizers.end(), metric) != optimizers.end()) {
```

Comment:
nit: `if (optimizers.find(metric) != optimizers.end()){...}` (same for `IsPhaseTimingMetric`)